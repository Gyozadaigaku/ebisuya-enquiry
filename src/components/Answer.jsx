import { createStyles, makeStyles } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";
import React from "react";

const useStyles = makeStyles(() =>
  createStyles({
    button: {
      borderColor: "#C14E49",
      color: "#C14E49",
      fontWeight: 600,
      marginBottom: "8px",
      "&:hover": {
        backgroundColor: "#C14E49",
        color: "white",
      },
    },
  })
);

const Answer = (props) => {
  const classes = useStyles();
  return (
    <Button
      className={classes.button}
      onClick={() => props.select(props.content, props.nextId)}
      variant="outlined"
    >
      {props.content}
    </Button>
  );
};

export default Answer;
