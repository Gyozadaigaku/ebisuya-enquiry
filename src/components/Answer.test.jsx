import Answer from "./Answer";
import renderer from "react-test-renderer";
import React from "react";

const selectAnswer = () => {};

test("matches to the snapshot", () => {
  const component = renderer.create(
    <Answer
      content={"予約したい"}
      key={1}
      nextId={"makeReservation"}
      select={selectAnswer}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
