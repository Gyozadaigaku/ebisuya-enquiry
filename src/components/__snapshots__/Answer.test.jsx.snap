// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-1"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    予約したい
  </span>
</button>
`;

exports[`matches to the snapshot 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-1"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    予約したい
  </span>
</button>
`;
